numba/transforms.py

```diff
@@ -53,8 +53,16 @@ def _loop_lift_get_candidate_infos(cfg, blocks, livemap):
     loopinfos = []
     for loop in loops:
         [callfrom] = loop.entries   # requirement checked earlier
-        an_exit = next(iter(loop.exits))  # anyone of the exit block
-        [(returnto, _)] = cfg.successors(an_exit)  # requirement checked earlier
+        # This loop exists to handle missing cfg.successors, only *an* exit is
+        # needed. Walk in stable order, higher exits first.
+        for an_exit in iter(sorted(loop.exits)):
+            # requirement checked earlier
+            ret = [x for x in cfg.successors(an_exit)]
+            if ret:
+                break
+        else:
+            continue # drop this loop from being liftable 
```

Comment:
Since the return statement in loop is not being handled by the looplifting logic, we shouldn't allow it for looplifting.  We should make it illegal to looplift if the loop contains a return statement.